pkg/manager/member/orphan_pods_cleaner.go

```diff
@@ -18,17 +18,33 @@ import (
 	"github.com/pingcap/tidb-operator/pkg/apis/pingcap.com/v1alpha1"
 	"github.com/pingcap/tidb-operator/pkg/controller"
 	"github.com/pingcap/tidb-operator/pkg/label"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 	corelisters "k8s.io/client-go/listers/core/v1"
 )
 
 const (
-	skipReasonOrphanPodsCleanerIsNotPDOrTiKV  = "orphan pods cleaner: member type is not pd or tikv"
-	skipReasonOrphanPodsCleanerPVCNameIsEmpty = "orphan pods cleaner: pvcName is empty"
-	skipReasonOrphanPodsCleanerPVCIsFound     = "orphan pods cleaner: pvc is found"
+	skipReasonOrphanPodsCleanerIsNotPDOrTiKV   = "orphan pods cleaner: member type is not pd or tikv"
+	skipReasonOrphanPodsCleanerPVCNameIsEmpty  = "orphan pods cleaner: pvcName is empty"
+	skipReasonOrphanPodsCleanerPVCIsFound      = "orphan pods cleaner: pvc is found"
+	skipReasonOrphanPodsCleanerPodIsNotPending = "orphan pods cleaner: pod is not pending"
 )
 
 // OrphanPodsCleaner implements the logic for cleaning the orphan pods(has no pvc)
+//
+// In scaling out and failover, we will try to delete the old PVC to prevent it
+// from being used by the new pod. However, the PVC might not be deleted
+// immediately in the apiserver because of finalizers (e.g.
+// kubernetes.io/pvc-protection) and the statefulset controller may not have
+// received PVC delete event when it tries to create the new replica.
+// Statefulset controller will create the new pod which will be pending forever
+// because no PVC to use. We need to clean these orphan pods and let the
```

Comment:
I think this description is not very accurate. When sts judges whether to create a replica, it will first determine whether the PVC of the pod already exists. If PVC exists, it will create a pod. But there is a case when sts judges that PVC already exists and this PVC actually is being deleted. When the pod is created, PVC has been deleted, causing the pod pending forever.

Reply:
>  But there is a case when sts judges that PVC already exists and this PVC actually is being deleted

I think `the statefulset controller may not have received PVC delete event when it tries to create the new replica` covers this case. The Delete event here is apiserver's object Delete event, not `deletionTimestamp=<time>` update event.